src/service/processors.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+from typing import List
+
+from xxhash import xxh32_intdigest
+
+
+def _get_partition_idx(word: str, num_partitions: int) -> int:
+    return xxh32_intdigest(word) % num_partitions
+
+
+def process_map_task(input_path: Path, num_partitions: int, output_dir: Path) -> List[Path]:
+    """
+    Processes the map task. Since it's output is split into num_partitions files, it
+    returns the output paths in a list of paths (in ascending order with regard to the partition num).
+
+    Each of the num_partitions output files will have n lines of form
+
+    "word 1"
```

Comment:
Wouldn't we want the mapper to produces files like "word number_of_times_word_got_encountered"? That would reduce the size of intermediate files and workload of the reducer.

Reply:
You're right, I was trying to follow the example we've seen. This is a good idea for now, however if were to drop the assumption about file contents being able to fit in memory, **technically** keeping a map with counts could blow up.
I discussed this assumption above. For now I'll change this in a follow up.

Reply:
I think we can keep the map words - counts in memory, at least for now. Even later, if we implement the splitter it still shouldn't overload the machines, right?